s30900-Pawel_Kozlowski/Zadania/14.10.2023 Procesy i Strumienie Zadania.txt

```diff
@@ -0,0 +1,387 @@
+
+   ps -> 
+   osboxes  ~  ps
+   PID TTY          TIME CMD
+   2486 pts/0    00:00:00 bash
+   3155 pts/0    00:00:00 ps
+   (Polecenia „ps” bez żadnej opcji wyświetla jedynie listy uruchomionych procesów z bieżącej powłoki)
+
+   ps -a ->
+   osboxes  ~  ps -a
+   PID TTY          TIME CMD
+   3209 pts/0    00:00:00 ps
+   (Używając opcji „ps -a” możemy wyświetlić wszystkie procesy z wyjątkiem liderów sesji i procesów, które nie są powiązane z  
+   terminalem.)
+
+   ps x ->
+   osboxes  ~  ps x
+   PID TTY      STAT   TIME COMMAND
+   1985 ?        Ss     0:00 /lib/systemd/systemd --user
+   1992 ?        S      0:00 (sd-pam)
+   1997 ?        Ss     0:00 /usr/bin/python3 /usr/bin/powerline-daemon --foregr
+   1998 ?        S<sl   0:00 /usr/bin/pulseaudio --daemonize=no --log-target=jou
+   1999 ?        Ss     0:00 /bin/bash /usr/local/bin/xfce4-session
+   2011 ?        Ss     0:00 /usr/bin/dbus-daemon --session --address=systemd: -
+   2080 ?        S      0:00 /usr/bin/VBoxClient --clipboard
+   2081 ?        Sl     0:00 /usr/bin/VBoxClient --clipboard
+   (...)
+
+   (Opcja x nakazuje ps wylistowanie procesów bez terminala sterującego. Są to głównie procesy uruchamiane podczas startu 
+   systemu i działające w tle .)
+
+   ps axu ->
+   osboxes  ~  ps aux
+   USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
+   root           1  0.0  0.2 102004 11348 ?        Ss   12:41   0:01 /sbin/init sp
+   root           2  0.0  0.0      0     0 ?        S    12:41   0:00 [kthreadd]
+   root           3  0.0  0.0      0     0 ?        I<   12:41   0:00 [rcu_gp]
+   root           4  0.0  0.0      0     0 ?        I<   12:41   0:00 [rcu_par_gp]
+   root           5  0.0  0.0      0     0 ?        I    12:41   0:01 [kworker/0:0-
+   root           6  0.0  0.0      0     0 ?        I<   12:41   0:00 [kworker/0:0H
+   root           9  0.0  0.0      0     0 ?        I<   12:41   0:00 [mm_percpu_wq
+   root          10  0.0  0.0      0     0 ?        S    12:41   0:00 [ksoftirqd/0]
+   root          11  0.0  0.0      0     0 ?        I    12:41   0:00 [rcu_sched]
+   (...)
+   (Opcja "a" anakazuje ps wyświetlenie procesów wszystkich użytkowników. Nie są pokazane tylko procesy 
+   niezwiązane   z terminalem i procesy liderów grup.)
+
+   "u" oznacza format zorientowany na użytkownika, który dostarcza szczegółowych informacji o procesach.
+
+   Opcja "x" nakazuje ps wylistowanie procesów bez terminala sterującego. Są to głównie procesy 
+   uruchamiane  podczas startu systemu i działające w tle .
+
+   Polecenie wyświetla informacje w jedenastu kolumnach oznaczonych USER, PID, %CPU, %MEM, VSZ, , , ,)
+
+   Jak myślisz, co oznacza znak zapytania w kolumnie 2?(wiedziałem)
+   - dany proces nie ma nadanego terminala
+   - różnica z x lub bez niego jest taka że: bez niego to wyuświetla wszystkie które mają terminal
+
+   B: Jak wykonasz poniższe 2 zadania?
+
+   Wyświetl wszystkie procesy bash ->
+   osboxes  ~  ps axu | grep bash
```

Comment:
Ta linia wyświetla wszystkie wiersze zawierające słowo bash.
Jeśli bedzie user o loginie `barabash` to jego wszystkie procesy niezależnie od nazwy zostaną wyświetlone.